commit_test.go

```diff
@@ -62,6 +63,47 @@ func (s *SuiteCommit) TestParents(c *C) {
 	c.Assert(output, DeepEquals, expected)
 }
 
+func (s *SuiteCommit) TestCommitEncodeDecodeIdempotent(c *C) {
+	ts, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05-07:00")
+	c.Assert(err, IsNil)
+	commits := []*Commit{
+		&Commit{
+			Author:    Signature{Name: "Foo", Email: "foo@example.local", When: ts},
+			Committer: Signature{Name: "Bar", Email: "bar@example.local", When: ts},
+			Message:   "Message\n\nFoo\nBar\nBaz\n\n",
+			tree:      core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			parents:   []core.Hash{core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d")},
+		},
+		&Commit{
+			Author:    Signature{Name: "Foo", Email: "foo@example.local", When: ts},
+			Committer: Signature{Name: "Bar", Email: "bar@example.local", When: ts},
+			Message:   "Message\n\nFoo\nBar\nBaz\n\n",
+			tree:      core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			parents: []core.Hash{
+				core.NewHash("b029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("c029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("d029517f6300c2da0f4b651b8642506cd6aaf45d"),
+				core.NewHash("e029517f6300c2da0f4b651b8642506cd6aaf45d"),
+			},
+		},
+	}
+	for _, commit := range commits {
+		// Compute hash
+		obj := &core.MemoryObject{}
+		err = commit.Encode(obj)
+		c.Assert(err, IsNil)
+		commit.Hash = obj.Hash()
```

Comment:
In order to know the exact object that we expect from Decode(Encode(Commit)), we need to know the hash of the encoded commit. If we use a git repo fixture as with other tests, this would be a non-issue, since we'd test Encode(Decode(Object)) and the Object hash would be one actually generated by git.

In order to do it in reverse, we need to encode the object to obtain the hash.

I'll change this to be a bit more concise though.